python/rmm/rmm.py

```diff
@@ -38,32 +37,91 @@ def _array_helper(addr, datasize, shape, strides, dtype, finalizer=None):
 
 
 # API Functions
-def initialize():
+def _initialize(
+    pool_allocator=False,
+    managed_memory=False,
+    initial_pool_size=None,
+    devices=0,
+    logging=False,
+):
     """
-    Initializes the RMM library using the options set in the librmm_config
-    module
+    Initializes RMM library using the options passed
     """
     allocation_mode = 0
-    if rmm_config.use_pool_allocator:
+
```

Comment:
This can be drastically simplified:

```suggestion
if pool_allocator:
    allocation_mode |= PoolAllocation
if managed_memory:
    allocation_mode |= CudaManagedMemory
```

Then you can delete lines 52-66.

Reply:
@harrism I considered this but felt like defining them explicitly was clearer (along with adding the additional enum option). If we're going back to ORing the enums then this can change accordingly as well.